ale/formatters/usgscsm_formatter.py

```diff
@@ -0,0 +1,91 @@
+import json
+from ale.base.type_sensor import LineScanner, Framer
+
+def to_usgscsm(driver):
+    isd_data = {}
+
+    # exterior orientation
+    body_radii = driver.target_body_radii
+    isd_data['radii'] = {
+        'semimajor' : body_radii[0],
+        'semiminor' : body_radii[1],
+        'unit' : 'm'
+    }
+    positions, velocities, position_times = driver.positions
+    isd_data['sensor_position'] = {
+        'positions' : positions,
+        'velocities' : velocities,
+        'unit' : 'm'
+    }
+    sun_positions, sun_velocities, _ = driver.sun_positions
+    isd_data['sun_position'] = {
+        'positions' : sun_positions,
+        'velocities' : sun_velocities,
+        'unit' : 'm'
+    }
+    rotation_chain = driver.rotation_chain
+    sensor_to_target = rotation_chain.rotation(rotation_chain[-1], rotation_chain[0])
+    quaternions, _, rotation_times = sensor_to_target.simplified_rotation()
+    isd_data['sensor_orientation'] = {
+        'quaternions' : quaternions
+    }
+
+    # interior orientation
+    isd_data['detector_sample_summing'] = driver.sample_summing
+    isd_data['detector_line_summing'] = driver.line_summing
+    isd_data['focal_length_model'] = {
+        'focal_length' : driver.focal_length
+    }
+    isd_data['detector_center'] = {
+        'line' : driver.detector_center_line,
+        'sample' : driver.detector_center_sample
+    }
+    isd_data['starting_detector_line'] = driver.starting_detector_line
+    isd_data['starting_detector_sample'] = driver.starting_detector_sample
+    isd_data['focal2pixel_lines'] = driver.focal2pixel_lines
+    isd_data['focal2pixel_samples'] = driver.focal2pixel_samples
+    isd_data['optical_distortion'] = driver.usgscsm_distortion_model
+
+    # general information
+    isd_data['image_lines'] = driver.line_count
+    isd_data['image_samples'] = driver.sample_count
+    isd_data['name_platform'] = driver.platform_name
+    isd_data['name_sensor'] = driver.sensor_name
+    isd_data['reference_height'] = {
+        "maxheight": 1000,
+        "minheight": -1000,
+        "unit": "m"
+    }
```

Comment:
I still don't know what reasonable values are for these. Should they be a percentage of the radii?

Reply:
What is this used for on the CSM side?

Reply:
This is used to support part of the CSM API:

> The getValidHeightRange () function returns the minimum and maximum ground heights that describe the range of validity of the model. For example, the model may not be valid at ground heights above the height of the sensor for physical models. This method is used in conjunction with getValidImageRange () to determine the full range of applicability of the sensor model.

Based on this, we should probably make these values as permissive as possible